api/src/paths/xlsx/transform.ts

```diff
@@ -0,0 +1,260 @@
+        // );
+        // await insertSubmissionMessage(
+        //   submissionStatusId,
+        //   'Error',
+        //   `Unable to fetch an appropriate transformation schema for your submission`,
+        //   'Missing Validation Schema',
+        //   connection
+        // );
+        // await connection.commit();
+        return res.status(200).json({ status: 'failed' });
+      }
+
+      req['transformationSchema'] = transformationSchema;
+
+      next();
+    } catch (error) {
+      defaultLog.debug({ label: 'getTransformationSchema', message: 'error', error });
+      await connection.rollback();
+      throw error;
+    } finally {
+      connection.release();
+    }
+  };
+}
+
+export function getTransformationRules(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'getTransformationRules', message: 'xlsx transform' });
+
+    try {
+      const transformationSchema: JSON = req['transformationSchema'];
+
+      const transformationSchemaParser = new TransformationSchemaParser(transformationSchema);
+
+      req['transformationSchemaParser'] = transformationSchemaParser;
+
+      next();
+    } catch (error) {
+      defaultLog.debug({ label: 'getTransformationRules', message: 'error', error });
+      throw error;
+    }
+  };
+}
+
+function transformXLSX(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'transformXLSX', message: 'xlsx transform' });
+
+    try {
+      const xlsxCsv: XLSXCSV = req['xlsx'];
+
+      const transformationSchemaParser: TransformationSchemaParser = req['transformationSchemaParser'];
+
+      const xlsxTransformation = new XLSXTransformation(transformationSchemaParser, xlsxCsv);
+
+      const transformedData = xlsxTransformation.transform();
+
+      const worksheets = xlsxTransformation.dataToSheet(transformedData);
+
+      const fileBuffers = Object.entries(worksheets).map(([fileName, worksheet]) => {
+        return {
+          name: fileName,
+          buffer: xlsxCsv.worksheetToBuffer(worksheet)
+        };
+      });
+
+      req['fileBuffers'] = fileBuffers;
+
+      next();
+    } catch (error) {
+      defaultLog.debug({ label: 'transformXLSX', message: 'error', error });
+      throw error;
+    }
+  };
+}
+
+export function persistTransformationResults(): RequestHandler {
+  return async (req, res, next) => {
+    try {
+      defaultLog.debug({ label: 'persistTransformationResults', message: 'xlsx transform' });
+
+      const fileBuffers: { name: string; buffer: Buffer }[] = req['fileBuffers'];
+
+      // Build the archive zip file
+      const dwcArchiveZip = new AdmZip();
+      fileBuffers.forEach((file) => dwcArchiveZip.addFile(`${file.name}.csv`, file.buffer));
+
+      // Build output s3Key based on the original input s3Key
+      const s3Key: string = req['s3Key'];
+      const outputS3KeyPrefix = s3Key.split('/').slice(0, -1).join('/'); // Remove the filename from original s3Key
```

Comment:
could you add a comment with an example?
```
/project/1/blah.txt -> /project/1
```

Reply:
Added